flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2428,7 +2430,8 @@ genTaskOp(lower::AbstractConverter &converter, lower::SymMap &symTable,
           ConstructQueue::const_iterator item) {
   lower::StatementContext stmtCtx;
```

Comment:
While working on the next patch in this series I've noticed that this `stmtCtx` seems not to be scoped correctly. If any cleanup code is added to it, it will emit that cleanup code when it is destroyed. Unfortunately, when this goes out of scope at the end of `genTaskOp` the builder insertion point is inside of the task operation. It would make more sense to do the cleanup after the end of the task operation to avoid adding unnecessary shared variables (this might be only a theoretical bug because I haven't yet found an example where any cleanup code is still present by the time we reach FIR.)

Reply:
I am trying to understand the problem here. This `stmtCtx` is used in managing and cleaning up intermediates while generating expressions of certain clauses. At the time the cleanup code is emitted, the insertion point is inside the task's region. Is it after the operations in task's region are generated, or before that?

Reply:
Let's discuss this on the PR for this change. Hopefully the example will make it clearer: https://github.com/llvm/llvm-project/pull/133891#issuecomment-2768860434